### PR TITLE
feat: Expose installed binaries under `PATH` for custom packages

### PR DESCRIPTION
**Pull request by isidentical**
Requires #44 first!

### CODE REVIEW

File: src/isolate/connections/_local/_base.py [hunk 1]
```diff
@@ -98,13 +120,22 @@ def get_env_vars(self) -> Dict[str, str]:
         custom_vars = {}
         custom_vars[AGENT_SIGNATURE] = "1"
         custom_vars["PYTHONUNBUFFERED"] = "1"
+
+        # NOTE: we don't have to manually set PYTHONPATH here if we are
+        # using a single environment since python will automatically
+        # use the proper path.
```
Comment (chamini2) [inline src/isolate/connections/_local/_base.py:126]: ATL: curious what happens if we just run it for when we are using a single environment?

**Review by chamini2 — COMMENTED** [review]
ATL: So this is to enable users to have access to any system package in the environments created? Normally added through conda

File: src/isolate/connections/_local/_base.py (hunk 1 above)
Comment (isidentical) [reply src/isolate/connections/_local/_base.py:126]: It shouldn't change anything, in an ideal case.

Comment (isidentical) [reply src/isolate/connections/_local/_base.py:126]: Seems like the only thing it does is shuffle the position (e.g. site packages [3rd party packages] goes last in the import path when you start it by default and now they go at the beginning).

```py
➜  /tmp ./test/bin/python -c "import sys; print(*enumerate(sys.path), sep='\n')"
(0, '')
(1, '/Users/isidentical/.pyenv/versions/3.9.11/lib/python39.zip')
(2, '/Users/isidentical/.pyenv/versions/3.9.11/lib/python3.9')
(3, '/Users/isidentical/.pyenv/versions/3.9.11/lib/python3.9/lib-dynload')
(4, '/private/tmp/test/lib/python3.9/site-packages')

➜  /tmp PYTHONPATH=/private/tmp/test/lib/python3.9/site-packages ./test/bin/python -c "import sys; print(*enumerate(sys.path), sep='\n')"
(0, '')
(1, '/private/tmp/test/lib/python3.9/site-packages')
(2, '/Users/isidentical/.pyenv/versions/3.9.11/lib/python39.zip')
(3, '/Users/isidentical/.pyenv/versions/3.9.11/lib/python3.9')
(4, '/Users/isidentical/.pyenv/versions/3.9.11/lib/python3.9/lib-dynload')```

Unless you install package with the same name as the stdlib one (most of them are blacklisted on PyPI), I don't think it is going to change anything. But we should probably fix it in the future for all PYTHONPATH based uses.

Comment (chamini2) [reply src/isolate/connections/_local/_base.py:126]: And should we do it then for single envs too to be consistent?

Comment (isidentical) [reply src/isolate/connections/_local/_base.py:126]: I think it makes sense to try to fix the multi-environments first (and if that doesn't work out, do it consistently for single environments), tracked in #46.

**Pull request merged**